hive/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java

```diff
@@ -0,0 +1,192 @@
+        !catalog.namespaceExists(Namespace.of("db2", "db2", "ns2")));
+  }
+
+  @Test
+  public void testSetNamespaceProperties() throws TException {
+    Namespace namespace = Namespace.of("dbname_set");
+
+    catalog.createNamespace(namespace, meta);
+    catalog.setProperties(namespace,
+        ImmutableMap.of(
+            "owner", "alter_apache",
+            "test", "test",
+            "location", "file:/data/tmp",
+            "comment", "iceberg test")
+    );
+
+    Database database = metastoreClient.getDatabase(namespace.level(0));
+    Assert.assertEquals(database.getParameters().get("owner"), "alter_apache");
+    Assert.assertEquals(database.getParameters().get("test"), "test");
+    Assert.assertEquals(database.getParameters().get("group"), "iceberg");
+    AssertHelpers.assertThrows("Should fail to namespace not exist" + namespace,
+        org.apache.iceberg.exceptions.NoSuchNamespaceException.class,
+        "Namespace does not exist: ", () -> {
+          catalog.setProperties(Namespace.of("db2", "db2", "ns2"), meta);
+        });
+  }
+
+  @Test
+  public void testRemoveNamespaceProperties() throws TException {
+    Namespace namespace = Namespace.of("dbname_remove");
+
+    catalog.createNamespace(namespace, meta);
+
+    catalog.removeProperties(namespace, ImmutableSet.of("comment", "owner"));
+
+    Database database = metastoreClient.getDatabase(namespace.level(0));
+
+    Assert.assertEquals(database.getParameters().get("owner"), null);
+    Assert.assertEquals(database.getParameters().get("group"), "iceberg");
+    AssertHelpers.assertThrows("Should fail to namespace not exist" + namespace,
+        org.apache.iceberg.exceptions.NoSuchNamespaceException.class,
+        "Namespace does not exist: ", () -> {
+          catalog.removeProperties(Namespace.of("db2", "db2", "ns2"), ImmutableSet.of("comment", "owner"));
+        });
+  }
+
+  @Test
+  public void testDropNamespace() throws TException {
+    Namespace namespace = Namespace.of("dbname_drop");
+    TableIdentifier identifier = TableIdentifier.of(namespace, "table");
+    Schema schema = new Schema(Types.StructType.of(
+        required(1, "id", Types.LongType.get())).fields());
+
+    catalog.createNamespace(namespace, meta);
+    catalog.createTable(identifier, schema);
+    Map<String, String> nameMata = catalog.loadNamespaceMetadata(namespace);
+    Assert.assertTrue(nameMata.get("owner").equals("apache"));
+    Assert.assertTrue(nameMata.get("group").equals("iceberg"));
+
+    AssertHelpers.assertThrows("Should fail to drop namespace is not empty" + namespace,
+        org.apache.iceberg.exceptions.NamespaceNotEmptyException.class,
+        "Namespace dbname_drop is not empty. One or more tables exist.", () -> {
+          catalog.dropNamespace(namespace);
+        });
+    Assert.assertTrue(catalog.dropTable(identifier, true));
+    Assert.assertTrue("Drop namespace " + namespace + " error ", catalog.dropNamespace(namespace));
```

Comment:
What is meant by "error" in this context message?

Reply:
fixed